packages/react-ui/src/index.d.ts

```diff
@@ -29,8 +29,8 @@ import Typography from './components/atoms/Typography';
 import Button from './components/atoms/Button';
 import PasswordField from './components/atoms/PasswordField';
 import SelectField from './components/atoms/SelectField';
-import UploadField from './components/molecules/UploadField';
-import AppBar from './components/organisms/AppBar';
+import UploadField from './components/molecules/UploadField/UploadField';
+import AppBar from './components/organisms/AppBar/AppBar';
```

Comment:
What about what we were discussing, to have an `index.js` instead of `UploadField/UploadField.js `?

Reply:
Please don't use `index.js` or any other convention that creates dozens of files with same name.
It is really hard to navigate in project, where half of files are named `index.js` or `main.yml`.
UploadField.js is much better imo.

(let alone fact, that single file component doesn't isn't enough reason to create _folder_ for it ...), why not having just`molecules/UploadField.js`  ?

`index.js` as concept also isn't ESM friendly, as ESM which is more or less future of JS completly ditched concept of "automatically" resolved "index" files that is CommonJS remnant.

Reply:
Happy to talk about this with more calm in a coffee, considering pros / cons. 

I wouldn't discard index.js / .ts as a good way to structure external visibility / exports of components / functions (while keeping internally, not exported, auxiliary files for the main component), keeping the main component in its own file. I think that's a discussion worth to have.

@zbigg if you have references / articules related to ESM about this topic, I will happily review them

Reply:
deck.gl is doing some work related to ESM & keeping index.js, eg here: https://github.com/visgl/deck.gl/blob/master/modules/arcgis/src/index.js